vortex-expr/src/binary.rs

```diff
@@ -44,7 +44,7 @@ impl VortexExpr for BinaryExpr {
         self
     }
 
-    fn evaluate(&self, batch: &ArrayData) -> VortexResult<ArrayData> {
+    fn unchecked_evaluate(&self, batch: &ArrayData) -> VortexResult<ArrayData> {
```

Comment:
What is this name? It's weird

Reply:
Sorry dan, I asked for this but now I don't think its needed

Reply:
Ah, I see what you're trying to do. I think it's tricky to enforce things like this without a wrapper type. e.g. `Expr(Arc<dyn ExprImpl>)`, that way you can expose an `Expr::evaluate` API that can call into things on `ExprImpl`. It's not a terrible idea actually..